.github/actions/install-wsl/action.yaml

```diff
@@ -0,0 +1,59 @@
+name: Install WSL
+author: Zygmunt Krynicki
+description: |
+  Install or upgrade WSL to the specified release.
+inputs:
+  wsl-msi-url:
+    description: URL of the MSI installer
+    required: false
+    default: https://github.com/microsoft/WSL/releases/download/2.2.4/wsl.2.2.4.0.x64.msi
+  wsl-msi-file:
+    description: Name of the WSL installer
+    required: false
+    default: wsl.2.2.4.0.x64.msi
+branding:
+  icon: download
+  color: orange
+runs:
+  using: composite
+  steps:
+    - name: Cache ${{ inputs.wsl-msi-file }} WSL installer
+      id: cache-wsl-installer
+      uses: actions/cache@v4
+      with:
+        path: ${{ inputs.wsl-msi-file }}
+        key: ${{ inputs.wsl-msi-file }}
+    - name: Download WSL installer version ${{ inputs.wsl-msi-file }}
+      if: ${{ steps.cache-wsl-installer.outputs.cache-hit != 'true' }}
+      shell: pwsh
+      run: |
+        Set-StrictMode -version latest
+        $ProgressPreference = 'SilentlyContinue'
+        Write-Output "Downloading WSL installer release from GitHub..."
+        Invoke-WebRequest -Uri "${{ inputs.wsl-msi-url }}" -OutFile "${{ inputs.wsl-msi-file }}" -UseBasicParsing
+        if ( ! $? ) {
+          exit 1
+        }
```

Comment:
since this is the last command in the script, won't it exit anyway if the status code is not 0?
I'm just wondering if this `if` check can be omitted.

Reply:
Not as far as I know. Powershell has two modes of execution (interactive and script) and the rules are somewhat complex/messy coming from UNIX world. I prefer to be as explicit as I can until I know for sure it is not required.